helix-core/src/main/java/org/apache/helix/util/HelixUtil.java

```diff
@@ -140,6 +160,82 @@ public static String serializeByComma(List<String> objects) {
     }
   }
 
+  /**
+   * Returns the expected ideal ResourceAssignments for the given resources in the cluster
+   * calculated using the read-only WAGED rebalancer.
+   * @param metadataStoreAddress
+   * @param clusterConfig
+   * @param instanceConfigs
+   * @param liveInstances
+   * @param newIdealStates
```

Comment:
The "new" here might be confusing. So the requirement here is, actually, the caller should include all the resource IS/Configs into the input, right?

Reply:
Yes. Let me rename them. I'll be removing "new".